src/Microsoft.DotNet.Darc/Darc/Operations/GatherDropOperation.cs

```diff
@@ -506,9 +510,20 @@ private void CreateReleasePackageLayout(List<DownloadedBuild> downloadedBuilds,
                     string relativePackagePath = Path.GetRelativePath(outputDirectory, targetFile);
                     allNupkgsFileContent.AppendLine(relativePackagePath);
 
-                    StringBuilder categoryStringBuilder = nupkgFileContents.GetOrAddValue(category,
-                        () => new StringBuilder());
-                    categoryStringBuilder.AppendLine(relativePackagePath);
+                    if(category == coreRepoCategory
+                        && packageFileName.Contains("Workload")
+                        && packageFileName.Contains("Manifest"))
+                    {
+                        StringBuilder categoryStringBuilder = nupkgFileContents.GetOrAddValue(manifestCategory,
+                            () => new StringBuilder());
+                        categoryStringBuilder.AppendLine(relativePackagePath);
+                    }
+                    else
+                    {
+                        StringBuilder categoryStringBuilder = nupkgFileContents.GetOrAddValue(category,
```

Comment:
The whole branch seems to differ only in one argument (category vs. manifestCategory) unless I'm missing something. Could you try to refactor the code so there's no duplication of the common parts?

Reply:
Updated it